WordPress/gutenberg
# Commands: update cmdk

Fixes #59703

## What?
This PR updates the command package cmdk from version `0.2.0` to `1.0.0`.

At the same time, it removes a temporary workaround for an issue that has been resolved upstream.

## Why?

- An issue with labels was reported in the cmdk library:  https://github.com/pacocoursey/cmdk/issues/196
- A temporary workaround was implemented to avoid this issue: https://github.com/WordPress/gutenberg/pull/56718
- The cmdk library has since fixed the issue and released it as version 1.0.0: https://github.com/pacocoursey/cmdk/pull/204

By updating cmdk to version `1.0.0`, this temporary workaround is no longer necessary.

## How?

Regarding updating the npm packages, I followed the instructions in the article, but please let me know if I'm wrong.

- In the `packages/commands/package.json` file, I manually updated the version of the `cmdk` line to `^1.0.0`.
- Run `npm install`

For labels, I removed the temporary workaround code. Instead, I added a new `label` prop that the List component now supports.


## Testing Instructions

- Open the command palette and enter some text.
- Open your browser developer tools and look at the HTML.
- Please make sure the HTML is properly constructed. The element with the `role="listbox"` attribute should still have the label `Command suggestions` (I've simplified the code for easier reading).

```html
<div class="commands-command-menu__container">
  <div tabindex="-1" cmdk-root="">
    <label cmdk-label="" for=":rg1:" id=":rg0:">Search commands and settings</label>
    <div class="commands-command-menu__header">
      <input placeholder="Search commands and settings" icon="" cmdk-input="" aria-controls=":rfv:" aria-labelledby=":rg0:" id=":rg1:" type="text">
    </div>
    <div cmdk-list="" role="listbox" aria-label="Command suggestions" id=":rfv:"> 
      <div cmdk-list-sizer="">
        <div cmdk-group="" role="presentation" data-value="undefined">
          <div cmdk-group-items="" role="group">
            <div id=":rg7:" cmdk-item="" role="option" aria-disabled="false" aria-selected="true" data-disabled="false" data-selected="true" data-value="Navigation">
              <div class="components-flex components-h-stack commands-command-menu__item has-icon">
                <span>Navigation</span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
```
